src/main/java/com/scottlogic/AuthorPostSorter.java

```diff
@@ -1,21 +1,25 @@
 package com.scottlogic;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 
 public class AuthorPostSorter implements PostSorter {
     @Override
     public List<UserPost> sort(List<UserPost> inputList, SortOrder sortOrder) {
+
         if (inputList == null) {
             return null;
         } else if (inputList.isEmpty()) {
             return inputList;
         }
-        Collections.sort(inputList, Comparator.comparing(UserPost::getAuthor));
-        if (sortOrder.equals(SortOrder.DESC)) {
-            Collections.reverse(inputList);
+        List<UserPost> userPosts = new ArrayList<UserPost>(inputList);
```

Comment:
be really clear with these list variable names, it's not obvious how inputList and userPosts differ, maybe it could be userPosts and sortedUserPosts. This makes it really clear that the second List<UserPost> is just a sorted version of the original list